### PR TITLE
Force the a11y focused node for a screen change notification on iOS.

### PR DESCRIPTION
**Pull request by darrenaustin**
Currently on iOS, when a screen change has occurred (a route change), we pass the most recent route scope's label as the parameter to the `UIAccessibilityScreenChangedNotification`.  VoiceOver then picks a new node to move its a11y focus to. The way it does this is seems pretty random, depending on the structure of the SemanticObjects we present it, and doesn't give us a lot of control. However, the parameter to `UIAccessibilityScreenChangedNotification` can also be the `SemanticObject` that we want to have focus after the change.

This PR makes use of this and passes the new route's label node as the focus for the screen change.  This fixes a couple of issues: https://github.com/flutter/flutter/issues/46625, and https://github.com/flutter/flutter/issues/44001. 

This seems like a good fix to me, but I am not that familiar with how our semantic tree deals with scopes. Can anyone see any issue with using the label node for focus instead of just announcing the label and letting VoiceOver pick the focus for us?

### CODE REVIEW

**Comment by dnfield** [comment]
Is there some way we could test this?

**Comment by darrenaustin** [comment]
> Is there some way we could test this?

I don't think we have any system in place for automated testing of this on iOS, so I am not sure what we can do for this.

I have run this manually against several test apps and the gallery. I didn't see anything that seemed broken to me, but that is certainly not exhaustive. Do we have a better way to evaluate these kinds of changes?

File: shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm [hunk 1]
```diff
@@ -224,20 +224,21 @@ - (BOOL)onCustomAccessibilityAction:(FlutterCustomAccessibilityAction*)action {
   return YES;
 }
 
-- (NSString*)routeName {
-  // Returns the first non-null and non-empty semantic label of a child
-  // with an NamesRoute flag. Otherwise returns nil.
+- (id)routeFocusObject {
+  // Returns the first SemanticObject in this branch that has
+  // the NamesRoute flag with a non-nil semantic label. Otherwise
+  // returns nil.
```
Comment (goderbauer) [inline shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: Is it guaranteed that a node with kNamesRoute is always focusable?

Comment (darrenaustin) [reply shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm]: I believe so. In the `isAccessibilityElement` method on `SemanticsObject` it uses:

```dart
  return (([self node].flags & ~flutter::kScrollableSemanticsFlags) != 0 &&
          [self node].flags != static_cast<int32_t>(flutter::SemanticsFlags::kIsHidden)) ||
         ![self node].label.empty() || ![self node].value.empty() || ![self node].hint.empty() ||
         ([self node].actions & ~flutter::kScrollableSemanticsActions) != 0;
```

So any node with a label should be a11y focusable.

**Comment by goderbauer** [comment]
@vick08 pointed out in https://github.com/material-components/material-components-flutter-gallery/issues/517 that the focus should always go to the first element on the screen. Maybe he can help with figuring if this is ok behavior.

**Comment by vick08** [comment]
Sorry. What I said in the mentioned issue is that accessibility services should manage the focus according to their specific needs. We should simply provide the necessary signals.
In this specific instance, we should be posting the notification that will place the focus on the first item on the screen. However, the developer should be able to override this parameter if they want to send the focus elsewhere.
Hope this helps!

**Comment by dnfield** [comment]
Closing in favor of https://github.com/flutter/engine/pull/19826

**Pull request closed without merging**